lib/puppetserver/ca/puppet_config.rb

```diff
@@ -75,19 +74,29 @@ def resolve_settings(overrides = {})
         cadir = overrides.fetch(:cadir, '$ssldir/ca')
         settings[:cadir] = substitutions['$cadir'] = cadir.sub(unresolved_setting, substitutions)
 
-        settings[:cacert] = overrides.fetch(:cacert, '$cadir/ca_crt.pem')
-        settings[:cakey] = overrides.fetch(:cakey, '$cadir/ca_key.pem')
-        settings[:cacrl] = overrides.fetch(:cacrl, '$cadir/ca_crl.pem')
-        settings[:serial] = overrides.fetch(:serial, '$cadir/serial')
+        # Note the call to hostname below, (tbd from Justin's code)
+        hostname = overrides.fetch(:certname, `hostname`.chomp)
+        settings[:certname] = substitutions['$certname'] = hostname.sub(unresolved_setting, substitutions)
+
+        settings[:ca_name] =        overrides.fetch(:ca_name, 'Puppet CA: $certname')
+        settings[:root_ca_name] =   overrides.fetch(:root_ca_name, 'Puppet Root CA: $certname')
```

Comment:
Thoughts on names?  Not sure if 'Puppet Root CA: $certname' is going to cut it.

Reply:
Any reason why not? (I don't really have a good feel for any conventions that might exist around this)

Reply:
Well i have absolutely no idea ;)